tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -393,16 +393,35 @@ final class TestingServer(
     textDocumentCapabilities.setFoldingRange(new FoldingRangeCapabilities)
     val initOptions = initializationOptions.getOrElse(
       InitializationOptions.Default.copy(
-        debuggingProvider = true,
-        treeViewProvider = true,
-        slowTaskProvider = true
+        debuggingProvider = Some(true),
+        treeViewProvider = Some(true),
+        slowTaskProvider = Some(true)
       )
     )
+
+    // Yes, this is a bit gross :/
+    // However, I want to only get the existing fields that are being set
+    // much like it'd be when a client actually sends this. This will just
+    // collect the fields that are set, get the values, and then make them into
+    // a map that will become a JsonObject to pass in as the InitializationOptions
+    val existingInitOptions =
```

Comment:
I would just add a simple json library for tests that takes into account options for tests, would be much nicer.

Reply:
Could you expand on what you mean by this?

Reply:
Ach, that was totally not cleat. I just meant adding something like upickle. Automatically transforming case class into a string and then using gson to parse that.

Reply:
So it looks like in order for uPickle to handle this correctly I would need to define a serializer into the companion object of `InitializationOptions`, but we don't include uPickle. I'm hesitant to add a library just to do this. What do you think?

Reply:
I haven't noticed that, I was thinking of only adding a library to the test project. This way we wouldn't be adding additional dependencies to the project itself.

We can leave it as is, it was just an idea.